osquery/tables/networking/posix/interfaces.cpp

```diff
@@ -187,7 +229,27 @@ void genDetailsFromAddr(const struct ifaddrs* addr, QueryData& results) {
     r["odrops"] = INTEGER(0);
     r["collisions"] = BIGINT_FROM_UINT32(ifd->ifi_collisions);
     r["last_change"] = BIGINT_FROM_UINT32(ifd->ifi_lastchange.tv_sec);
-#endif
+    if (context.isColumnUsed("link_speed")) {
+      int fd = socket(AF_INET, SOCK_DGRAM, 0);
+      if (fd >= 0) {
+        struct ifmediareq ifmr;
+        memset(&ifmr, 0, sizeof(ifmr));
+        memcpy(ifmr.ifm_name, addr->ifa_name, sizeof(ifmr.ifm_name));
+        // int *media_list = (int *)malloc(ifmr.ifm_count * sizeof(int));
+        const std::unique_ptr<int[]> media_list(new int[ifmr.ifm_count]);
```

Comment:
What is the magnitude of count? Would it make sense to just put it on the stack or could it be very large?